polkadot/node/core/pvf/common/src/error.rs

```diff
@@ -14,13 +14,21 @@
 // You should have received a copy of the GNU General Public License
 // along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
 
-use crate::prepare::PrepareStats;
+use crate::prepare::{PrepareStats, PrepareSuccess, PrepareWorkerSuccess};
 use parity_scale_codec::{Decode, Encode};
 use std::fmt;
 
-/// Result of PVF preparation performed by the validation host. Contains stats about the preparation
-/// if successful
-pub type PrepareResult = Result<PrepareStats, PrepareError>;
+/// Result of PVF preparation from a worker, with checksum of the compiled PVF and stats of the
+/// preparation if successful.
+pub type PrepareWorkerResult = Result<PrepareWorkerSuccess, PrepareError>;
+
+/// Result of PVF preparation propagated all the way back to the host, with path to the concluded
+/// artifact and stats of the preparation if successful.
+pub type PrepareResult = Result<PrepareSuccess, PrepareError>;
+
+/// Result of prechecking PVF performed by the validation host. Contains stats about the preparation
+/// if successful.
+pub type PrecheckResult = Result<PrepareStats, PrepareError>;
```

Comment:
I don't think the stats are used by the caller, can it just be `Result<(), PrepareError>`? (I haven't tried it, but *think* it should work.)

Reply:
I guess this is used for some response of a cli command right? I was just keeping it as it was before. The code should compile but Is it ok to change the response? Does it affect the user? I'll trust your judgement here. :)

Reply:
Good questions! And yeah, it was already the case that we returned the stats. Previously the stats were part of the overloaded `PrepareResult`, but now that we're using different results in different contexts, we can reconsider whether the stats are needed here.

So no, removing them here shouldn't affect the user at all. Prechecking is requested by the `pvf-checker` subsystem and the message is handled [here](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/node/core/candidate-validation/src/lib.rs#L230). We throw away the specific success value and simply [report the outcome](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/node/core/candidate-validation/src/lib.rs#L378-L386), whether prechecking succeeded or not. So since the stats are used only within the PVF host itself for metrics purposes, they don't need to be returned higher up. Doing so kind of implies that they are used by `pvf-checker`, when they are not.